src/Compilers/Core/CodeAnalysisTest/Symbols/NullablilityInfoTests.cs

```diff
@@ -16,15 +16,13 @@ public void Equality()
                                     new NullabilityInfo(NullableAnnotation.Annotated, NullableFlowState.NotNull),
                                     equal: true);
 
-#pragma warning disable IDE0055 // Fix formatting: spacing is intentional to allow for visual field comparison
-            assertEqualsAndHashCode(new NullabilityInfo(NullableAnnotation.Annotated,    NullableFlowState.NotNull),
+            assertEqualsAndHashCode(new NullabilityInfo(NullableAnnotation.Annotated, NullableFlowState.NotNull),
                                     new NullabilityInfo(NullableAnnotation.NotAnnotated, NullableFlowState.NotNull),
                                     equal: false);
 
             assertEqualsAndHashCode(new NullabilityInfo(NullableAnnotation.Annotated, NullableFlowState.MaybeNull),
                                     new NullabilityInfo(NullableAnnotation.Annotated, NullableFlowState.NotNull),
                                     equal: false);
-#pragma warning restore IDE0055 // Fix formatting
```

Comment:
@333fred it is not possible to suppress Format Document within a region of a document. The warning suppression here was ignored since the code inside it was automatically reformatted anyway.

Reply:
Assuming for a second that we needed to suppress IDE0055 in this area. How would that have been done properly.

---
In reply to: [321929178](https://github.com/dotnet/roslyn/pull/38562#discussion_r321929178) [](ancestors = 321929178)

Reply:
@jaredpar There is currently no mechanism for doing so. The only exclusion mechanism is marking the file with a `// <auto-generated/>` comment at the top which the dotnet-format tool and Code Cleanup knows to ignore.

Reply:
📝 The reason for this limitation is the dotnet-format tool does not "fix diagnostics". Rather, it runs Format Document on all the files, and that action doesn't understand diagnostics or suppressions.

Reply:
@jasonmalinowski thinks we may be asking and answering different questions here. 

What I'm asking is can I disable the IDE0055 diagnostic using the pattern @333fred did above? Essentially in a scenario where we had warning as error in CI, and there was a bug in our analyzer, could `#pragma warning disable IDE0055` be a tool to get my change through CI (while in parallel working on the fixing the analyzer).

If the answer is no to that then how would we disable the anazyler for a block of code, perhaps a whole file, to unblock developers? Basically what's my "break class in case of fire" for the formatting analyzer.

Reply:
@jaredpar Yes, you can suppress IDE0055 and it will not report a warning during the build. However, when any other user facing any other IDE0055 warning elsewhere in Roslyn attempts to format the files and fix them, the suppression will be ignored and the content will be formatted anyway. At that point, the suppression is no longer needed because the formatting was changed to match what the formatter was expecting.

Reply:
Should the formatter be taught to not format syntax within regions where IDE0055 has been pragma disabled? It seems odd for a code fix to fix suppressed diagnostics.

Reply:
We could, and it wouldn't be particularly difficult - just add a suppress formatting rule between the directives the same way we do for braces of an object initializer.

Reply:
➡️ #38587